GetStream/stream-chat-swift
# Fix Message List not updating when user info changes

### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/520

### 🎯 Goal
Fixes Message List not updating when user info changes.

### 🛠 Implementation
The root of the problem was that although we updated the UserDTO whenever there was a user-updated event, we also needed to trigger updates to the messages that that user owns. Besides this, we also did not account for the `message.author` in the diffing of the message list. 

### 🎨 Showcase

https://github.com/GetStream/stream-chat-swift/assets/12814114/93d26ada-e940-4961-a1dc-d8e39af65463


### 🧪 Manual Testing Notes

1. Open a Channel
2. Update the user data of one of the members.
    ```bash
    stream-cli chat upsert-user --properties "{\"id\":\"leia_organa\",\"name\":\"Leia\", \"image\":\"https://www.themarysue.com/wp-content/uploads/2017/08/leiatop1.jpg?resize=650%2C574\"}"
    ```
3. Should update all the messages with the new user info

**Note:** Don't forget to update the user back to the previous info
```bash
stream-cli chat upsert-user --properties "{\"id\":\"leia_organa\",\"name\":\"Leia Organa\", \"image\":\"https://vignette.wikia.nocookie.net/starwars/images/2/20/LukeTLJ.jpg\"}"
```

### ☑️ Contributor Checklist
